drivers/pn532/pn532.c

```diff
@@ -120,23 +124,21 @@ int pn532_init(pn532_t *dev, const pn532_params_t *params, pn532_mode_t mode)
     dev->mode = mode;
     if (mode == PN532_I2C) {
 #ifdef PN532_SUPPORT_I2C
-        ret = i2c_init_master(dev->conf->i2c, I2C_SPEED_FAST);
+        if (i2c_init_master(dev->conf->i2c, I2C_SPEED_FAST) != 0) {
+            DEBUG("pn532: initialization of I2C bus failed\n");
+            return -1;
+        }
 #endif
     }
     else {
 #ifdef PN532_SUPPORT_SPI
-        ret = spi_init_master(dev->conf->spi, SPI_CONF_FIRST_RISING,
```

Comment:
`ret` is initially set to `-1` and this function returns `ret` => `pn532_init` always returns `-1` this way and thusly fails